// @gimmeapps/gquicklib-angular/esm2022/lib/ui/helpers/base/base.component.mjs
import { Component, Input } from "@angular/core";
import * as i0 from "@angular/core";
export class BaseComponent {
    constructor() {
        this.color = "primary";
        this.componentStyle = "flat";
        this.componentSize = "medium";
        this.round = false;
        this.disabled = false;
    }
    ngOnInit() { }
    ngOnDestroy() { }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: BaseComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.0", type: BaseComponent, selector: "quick-base-component", inputs: { color: "color", componentStyle: "componentStyle", componentSize: "componentSize", round: "round", disabled: "disabled" }, ngImport: i0, template: '', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: BaseComponent, decorators: [{
            type: Component,
            args: [{
                    'selector': 'quick-base-component',
                    'template': ''
                }]
        }], propDecorators: { color: [{
                type: Input
            }], componentStyle: [{
                type: Input
            }], componentSize: [{
                type: Input
            }], round: [{
                type: Input
            }], disabled: [{
                type: Input
            }] } });
