// @gimmeapps/gquicklib-angular/esm2022/lib/modals/modals.module.mjs
import { NgModule } from "@angular/core";
import { ShadowboxComponent } from './shadowbox/shadowbox.component';
import { CommonModule } from "@angular/common";
import { ButtonModule } from "../ui";
import * as i0 from "@angular/core";
export class ModalsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ModalsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: ModalsModule, declarations: [ShadowboxComponent], imports: [CommonModule,
            ButtonModule], exports: [ShadowboxComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ModalsModule, imports: [CommonModule,
            ButtonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ModalsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ShadowboxComponent
                    ],
                    imports: [
                        CommonModule,
                        ButtonModule
                    ],
                    exports: [
                        ShadowboxComponent
                    ]
                }]
        }] });
