// @gimmeapps/gquicklib-angular/esm2022/lib/shared/services/rest.service.mjs
import { HttpParams } from "@angular/common/http";
import { FilterBuilder } from "../../../public-api";
export class RestService {
    constructor(http) {
        this.http = http;
    }
    getQuery(filter) {
        let params = new HttpParams();
        if (filter) {
            const builder = new FilterBuilder(filter);
            params = params.append("filter", builder.encode());
        }
        return params;
    }
    query(filter) {
        let params = this.getQuery(filter);
        return this.http.get(this.api, { params });
    }
    paginate(page, pageSize, filter) {
        let params = this.getQuery(filter);
        params = params.append("page", page);
        if (pageSize)
            params = params.append("pagesize", pageSize);
        return this.http.get(this.api, { params });
    }
    lookup(filter) {
        let params = this.getQuery(filter);
        return this.http.get(`${this.api}/lookup`, { params });
    }
    getById(id) {
        return this.http.get(`${this.api}/${id}`);
    }
    insert(dados) {
        return this.http.post(`${this.api}`, dados);
    }
    update(id, dados) {
        return this.http.post(`${this.api}/${id}`, dados);
    }
    delete(id) {
        return this.http.delete(`${this.api}/${id}`);
    }
    bulk(dados) {
        return this.http.post(`${this.api}/bulk`, dados);
    }
}
