// @gimmeapps/gquicklib-angular/esm2022/lib/file-manager/file-manager.module.mjs
import { NgModule } from "@angular/core";
import { PhotoGaleryComponent } from './photo-galery/photo-galery.component';
import { CommonModule } from "@angular/common";
import { ButtonModule } from "../ui";
import { ModalsModule } from "../modals";
import * as i0 from "@angular/core";
export class FileManagerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: FileManagerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: FileManagerModule, declarations: [PhotoGaleryComponent], imports: [CommonModule,
            ButtonModule,
            ModalsModule], exports: [PhotoGaleryComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: FileManagerModule, imports: [CommonModule,
            ButtonModule,
            ModalsModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: FileManagerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        PhotoGaleryComponent
                    ],
                    imports: [
                        CommonModule,
                        ButtonModule,
                        ModalsModule
                    ],
                    exports: [
                        PhotoGaleryComponent
                    ]
                }]
        }] });
