// @gimmeapps/gquicklib-angular/esm2022/lib/shared/helpers/filter-builder.mjs
export class FilterBuilder {
    constructor(filter) {
        filter = !filter ? { "and": [] } : JSON.parse(JSON.stringify(filter));
        if (this.isFilterCondition(filter))
            filter = { "and": [filter] };
        this.filter = filter;
    }
    insert(criteria) {
        const root = Object.keys(this.filter)[0];
        if (root === 'and') {
            this.filter.and.push(criteria);
        }
        else if (root === 'or') {
            this.filter.or.push(criteria);
        }
        else {
            throw new Error('Operador de grupo inválido.');
        }
    }
    insertMany(criteria) {
        criteria.forEach(c => this.insert(c));
    }
    encode() {
        const json = JSON.stringify(this.filter);
        return btoa(json);
    }
    isFilterCondition(obj) {
        return (obj.hasOwnProperty('=') ||
            obj.hasOwnProperty('<') ||
            obj.hasOwnProperty('<=') ||
            obj.hasOwnProperty('>') ||
            obj.hasOwnProperty('>=') ||
            obj.hasOwnProperty('isnull'));
    }
}
