// @gimmeapps/gquicklib-angular/esm2022/lib/forms/quick-forms.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SharedModule } from '../shared';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NgxMaskDirective, NgxMaskPipe, provideNgxMask } from 'ngx-mask';
import { FormFieldComponent } from './form-field/form-field.component';
import { FormAutocompleteComponent } from './form-field/form-autocomplete/form-autocomplete.component';
import { FormErrorComponent } from './form-field/form-error/form-error.component';
import * as i0 from "@angular/core";
export class QuickFormsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: QuickFormsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: QuickFormsModule, declarations: [FormFieldComponent,
            FormAutocompleteComponent,
            FormErrorComponent], imports: [CommonModule,
            SharedModule,
            FormsModule,
            ReactiveFormsModule,
            NgxMaskDirective,
            NgxMaskPipe], exports: [FormsModule,
            ReactiveFormsModule,
            FormFieldComponent,
            FormAutocompleteComponent,
            FormErrorComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: QuickFormsModule, providers: [
            provideNgxMask()
        ], imports: [CommonModule,
            SharedModule,
            FormsModule,
            ReactiveFormsModule, FormsModule,
            ReactiveFormsModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: QuickFormsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        FormFieldComponent,
                        FormAutocompleteComponent,
                        FormErrorComponent
                    ],
                    imports: [
                        CommonModule,
                        SharedModule,
                        FormsModule,
                        ReactiveFormsModule,
                        NgxMaskDirective,
                        NgxMaskPipe
                    ],
                    providers: [
                        provideNgxMask()
                    ],
                    exports: [
                        FormsModule,
                        ReactiveFormsModule,
                        FormFieldComponent,
                        FormAutocompleteComponent,
                        FormErrorComponent
                    ]
                }]
        }] });
