// @gimmeapps/gquicklib-angular/esm2022/lib/file-manager/photo-galery/photo-galery.component.mjs
import { Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../ui/button/button.component";
import * as i3 from "../../modals/shadowbox/shadowbox.component";
export class PhotoGaleryComponent {
    constructor() {
        this.disabled = false;
        this.onChange = new EventEmitter();
        this.onRemove = new EventEmitter();
        this.images = [];
        this.preview = 0;
    }
    onFileSelected(event) {
        const files = event.target.files;
        this.addImages(files);
    }
    onDragOver(event) {
        event.preventDefault();
    }
    onDrop(event) {
        event.preventDefault();
        const files = event.dataTransfer?.files || new FileList();
        this.addImages(files);
    }
    addImages(files) {
        const loadImage = (file) => {
            return new Promise(resolve => {
                const reader = new FileReader();
                reader.onload = (e) => {
                    resolve({ url: e.target.result, file });
                };
                reader.readAsDataURL(file);
            });
        };
        const tasks = [];
        for (let i = 0; i < files.length; i++) {
            const file = files[i];
            tasks.push(loadImage(file));
        }
        Promise.all(tasks).then(images => {
            this.images.push(...images);
            this.onChange.emit(this.images);
        });
    }
    removeImage(image) {
        const index = this.images.indexOf(image);
        if (index !== -1) {
            this.images.splice(index, 1);
            this.onRemove.emit(image);
            this.onChange.emit(this.images);
        }
    }
    openFilePicker() {
        this.fileInput?.nativeElement.click();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: PhotoGaleryComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.0", type: PhotoGaleryComponent, selector: "quick-photo-galery", inputs: { disabled: "disabled", images: "images" }, outputs: { onChange: "onChange", onRemove: "onRemove" }, viewQueries: [{ propertyName: "fileInput", first: true, predicate: ["fileInput"], descendants: true }], ngImport: i0, template: "<quick-shadowbox #shadowbox [images]=\"images\" [current]=\"preview\" />\n\n<div class=\"upload-images\">\n    @if(!disabled){<div class=\"drop-area\" (drop)=\"onDrop($event)\" (dragover)=\"onDragOver($event)\"\n        (click)=\"openFilePicker()\">\n        Arraste e solte as imagens aqui ou clique para selecionar arquivos\n        <input type=\"file\" multiple (change)=\"onFileSelected($event)\" #fileInput>\n    </div>}\n\n    <div class=\"thumbnails\">\n        <div *ngFor=\"let image of images; let i = index\" class=\"thumbnail\">\n            <img [src]=\"image.url\" alt=\"Imagem\" (click)=\"preview = i; shadowbox.show()\">\n            @if(!disabled){\n            <quick-button class=\"btn-delete\" icon=\"delete\" componentSize=\"small\" hint=\"Excluir imagem\" color=\"danger\"\n                [round]=\"true\" (onClick)=\"removeImage(image)\" />\n            }\n        </div>\n    </div>\n</div>", styles: [".upload-images{display:flex;flex-direction:column}.drop-area{border:2px dashed #ccc;padding:20px;cursor:pointer;font-weight:500;color:var(--gray-700);text-align:center;text-wrap:wrap}.thumbnails{display:flex;flex-wrap:wrap}.thumbnail{margin:10px;position:relative}.thumbnail img{max-width:100px;max-height:100px;cursor:pointer}input[type=file]{display:none}.btn-delete{position:absolute;margin:.2rem;right:0}\n"], dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "component", type: i2.ButtonComponent, selector: "quick-button", inputs: ["icon", "text", "hint"], outputs: ["onClick"] }, { kind: "component", type: i3.ShadowboxComponent, selector: "quick-shadowbox", inputs: ["images", "current"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: PhotoGaleryComponent, decorators: [{
            type: Component,
            args: [{ selector: 'quick-photo-galery', template: "<quick-shadowbox #shadowbox [images]=\"images\" [current]=\"preview\" />\n\n<div class=\"upload-images\">\n    @if(!disabled){<div class=\"drop-area\" (drop)=\"onDrop($event)\" (dragover)=\"onDragOver($event)\"\n        (click)=\"openFilePicker()\">\n        Arraste e solte as imagens aqui ou clique para selecionar arquivos\n        <input type=\"file\" multiple (change)=\"onFileSelected($event)\" #fileInput>\n    </div>}\n\n    <div class=\"thumbnails\">\n        <div *ngFor=\"let image of images; let i = index\" class=\"thumbnail\">\n            <img [src]=\"image.url\" alt=\"Imagem\" (click)=\"preview = i; shadowbox.show()\">\n            @if(!disabled){\n            <quick-button class=\"btn-delete\" icon=\"delete\" componentSize=\"small\" hint=\"Excluir imagem\" color=\"danger\"\n                [round]=\"true\" (onClick)=\"removeImage(image)\" />\n            }\n        </div>\n    </div>\n</div>", styles: [".upload-images{display:flex;flex-direction:column}.drop-area{border:2px dashed #ccc;padding:20px;cursor:pointer;font-weight:500;color:var(--gray-700);text-align:center;text-wrap:wrap}.thumbnails{display:flex;flex-wrap:wrap}.thumbnail{margin:10px;position:relative}.thumbnail img{max-width:100px;max-height:100px;cursor:pointer}input[type=file]{display:none}.btn-delete{position:absolute;margin:.2rem;right:0}\n"] }]
        }], propDecorators: { disabled: [{
                type: Input
            }], fileInput: [{
                type: ViewChild,
                args: ['fileInput']
            }], onChange: [{
                type: Output
            }], onRemove: [{
                type: Output
            }], images: [{
                type: Input
            }] } });
