// @gimmeapps/gquicklib-angular/esm2022/lib/ui/button/button.module.mjs
import { NgModule } from "@angular/core";
import { ButtonComponent } from "./button.component";
import { IconModule } from "../icon";
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
export class ButtonModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: ButtonModule, declarations: [ButtonComponent], imports: [CommonModule,
            IconModule], exports: [ButtonComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ButtonModule, imports: [CommonModule,
            IconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: ButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ButtonComponent
                    ],
                    imports: [
                        CommonModule,
                        IconModule
                    ],
                    exports: [
                        ButtonComponent
                    ]
                }]
        }] });
