// @gimmeapps/gquicklib-angular/esm2022/lib/inputs/inputs.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TextFieldComponent } from './text-field/text-field.component';
import { PickerComponent } from './picker/picker.component';
import { UiModule } from '../ui';
import { TextFieldSuffixComponent } from './text-field/text-field-suffix/text-field-suffix.component';
import { FormsModule } from '@angular/forms';
import { SharedModule } from '../shared';
import { StarRatingComponent } from './star-rating/star-rating.component';
import * as i0 from "@angular/core";
export class InputsModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: InputsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: InputsModule, declarations: [PickerComponent,
            TextFieldComponent,
            TextFieldSuffixComponent,
            StarRatingComponent], imports: [CommonModule,
            FormsModule,
            SharedModule,
            UiModule], exports: [PickerComponent,
            TextFieldComponent,
            TextFieldSuffixComponent,
            StarRatingComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: InputsModule, imports: [CommonModule,
            FormsModule,
            SharedModule,
            UiModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: InputsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        PickerComponent,
                        TextFieldComponent,
                        TextFieldSuffixComponent,
                        StarRatingComponent,
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        SharedModule,
                        UiModule
                    ],
                    exports: [
                        PickerComponent,
                        TextFieldComponent,
                        TextFieldSuffixComponent,
                        StarRatingComponent
                    ]
                }]
        }] });
