// @gimmeapps/gquicklib-angular/esm2022/lib/inputs/star-rating/star-rating.component.mjs
import { Component, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/flex-layout/extended";
import * as i3 from "../../ui/icon/icon.component";
export class StarRatingComponent {
    constructor() {
        this.disabled = false;
        this.mouseOver = null;
        this.rating = 0;
        this.maxRate = 5;
        this.color = "primary";
        this.onChange = (rating) => { };
        this.onTouched = () => { };
    }
    writeValue(obj) {
        this.rating = obj;
    }
    registerOnChange(onChange) {
        this.onChange = onChange;
    }
    registerOnTouched(onTouched) {
        this.onTouched = onTouched;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    onMouseEnter(rate) {
        if (this.disabled)
            return;
        this.mouseOver = rate;
    }
    onMouseLeave() {
        if (this.disabled)
            return;
        this.mouseOver = null;
    }
    setValue(value) {
        if (this.disabled)
            return;
        this.writeValue(value);
        this.onChange(value);
    }
    get stars() {
        const stars = new Array(this.maxRate).fill(0).map((value, index) => {
            return index + 1;
        });
        return stars;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: StarRatingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.0", type: StarRatingComponent, selector: "quick-star-rating", inputs: { maxRate: "maxRate", color: "color" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                multi: true,
                useExisting: StarRatingComponent
            }
        ], ngImport: i0, template: "<ul class=\"{{color}}\" [ngClass]=\"{'disabled':disabled}\">\n    @for (rate of stars; track $index) {\n    <li (click)=\"setValue(rate)\" (mouseenter)=\"onMouseEnter(rate)\" (mouseleave)=\"onMouseLeave()\"\n        [ngClass]=\"{'active': (mouseOver && mouseOver >= rate) || (!mouseOver && rating >= rate)}\">\n        <quick-icon iconName=\"star\" />\n    </li>\n    }\n</ul>", styles: ["ul{display:flex;flex-direction:row;align-items:center;height:42px;box-sizing:border-box;padding:0 .5rem;font-size:1.25rem}ul.primary{color:var(--primary-color)}ul.accent{color:var(--accent-color)}ul.info{color:var(--info-color)}ul.success{color:var(--success-color)}ul.warn{color:var(--warn-color)}ul.danger{color:var(--danger-color)}ul li{cursor:pointer;opacity:.2}ul li.active{opacity:1}ul.disabled{opacity:.8}\n"], dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i2.DefaultClassDirective, selector: "  [ngClass], [ngClass.xs], [ngClass.sm], [ngClass.md], [ngClass.lg], [ngClass.xl],  [ngClass.lt-sm], [ngClass.lt-md], [ngClass.lt-lg], [ngClass.lt-xl],  [ngClass.gt-xs], [ngClass.gt-sm], [ngClass.gt-md], [ngClass.gt-lg]", inputs: ["ngClass", "ngClass.xs", "ngClass.sm", "ngClass.md", "ngClass.lg", "ngClass.xl", "ngClass.lt-sm", "ngClass.lt-md", "ngClass.lt-lg", "ngClass.lt-xl", "ngClass.gt-xs", "ngClass.gt-sm", "ngClass.gt-md", "ngClass.gt-lg"] }, { kind: "component", type: i3.IconComponent, selector: "quick-icon", inputs: ["iconName"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: StarRatingComponent, decorators: [{
            type: Component,
            args: [{ selector: 'quick-star-rating', providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            multi: true,
                            useExisting: StarRatingComponent
                        }
                    ], template: "<ul class=\"{{color}}\" [ngClass]=\"{'disabled':disabled}\">\n    @for (rate of stars; track $index) {\n    <li (click)=\"setValue(rate)\" (mouseenter)=\"onMouseEnter(rate)\" (mouseleave)=\"onMouseLeave()\"\n        [ngClass]=\"{'active': (mouseOver && mouseOver >= rate) || (!mouseOver && rating >= rate)}\">\n        <quick-icon iconName=\"star\" />\n    </li>\n    }\n</ul>", styles: ["ul{display:flex;flex-direction:row;align-items:center;height:42px;box-sizing:border-box;padding:0 .5rem;font-size:1.25rem}ul.primary{color:var(--primary-color)}ul.accent{color:var(--accent-color)}ul.info{color:var(--info-color)}ul.success{color:var(--success-color)}ul.warn{color:var(--warn-color)}ul.danger{color:var(--danger-color)}ul li{cursor:pointer;opacity:.2}ul li.active{opacity:1}ul.disabled{opacity:.8}\n"] }]
        }], propDecorators: { maxRate: [{
                type: Input
            }], color: [{
                type: Input
            }] } });
