// @gimmeapps/gquicklib-angular/esm2022/lib/ui/ui.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SharedModule } from '../shared';
import { ToolbarComponent } from './toolbar/toolbar.component';
import { ColumnComponent, RowComponent } from './layout';
import { CardModule } from './card/card.module';
import { ButtonModule } from './button';
import { IconModule } from './icon';
import { DatagridComponent } from './datagrid';
import { FormsModule } from '@angular/forms';
import { RatingBadgeComponent } from './rating-badge/rating-badge.component';
import * as i0 from "@angular/core";
export class UiModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: UiModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.0", ngImport: i0, type: UiModule, declarations: [ColumnComponent,
            RowComponent,
            ToolbarComponent,
            DatagridComponent,
            RatingBadgeComponent], imports: [CommonModule,
            SharedModule,
            ButtonModule,
            FormsModule,
            IconModule], exports: [ButtonModule,
            ColumnComponent,
            CardModule,
            DatagridComponent,
            IconModule,
            RowComponent,
            RatingBadgeComponent,
            ToolbarComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: UiModule, imports: [CommonModule,
            SharedModule,
            ButtonModule,
            FormsModule,
            IconModule, ButtonModule,
            CardModule,
            IconModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: UiModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ColumnComponent,
                        RowComponent,
                        ToolbarComponent,
                        DatagridComponent,
                        RatingBadgeComponent
                    ],
                    imports: [
                        CommonModule,
                        SharedModule,
                        ButtonModule,
                        FormsModule,
                        IconModule
                    ],
                    exports: [
                        ButtonModule,
                        ColumnComponent,
                        CardModule,
                        DatagridComponent,
                        IconModule,
                        RowComponent,
                        RatingBadgeComponent,
                        ToolbarComponent,
                    ]
                }]
        }] });
