// @gimmeapps/gquicklib-angular/esm2022/lib/forms/form-field/form-autocomplete/form-autocomplete.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class FormAutocompleteComponent {
    constructor() {
        this.options = [];
        this.filterText = "";
        this.isVisible = false;
    }
    ngOnInit() {
        if (this.target) {
            this.target.autocomplete = "off";
            this.target.addEventListener('focus', this.onTargetFocus.bind(this));
            this.target.addEventListener('blur', this.onTargetBlur.bind(this));
            this.target.addEventListener('input', this.onTargetInput.bind(this));
        }
    }
    onTargetFocus(ev) {
        this.isVisible = true;
        if (this.target)
            this.filterText = this.target.value;
    }
    onTargetBlur(ev) {
        setTimeout(() => {
            this.isVisible = false;
        }, 150);
    }
    onTargetInput(ev) {
        if (this.target)
            this.filterText = this.target.value;
    }
    select(value) {
        if (this.control) {
            this.control.setValue(value);
        }
        else if (this.target) {
            this.target.value = value;
        }
    }
    get filteredOptions() {
        return this.options.filter(o => o.toLowerCase().includes(this.filterText.toLowerCase()));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: FormAutocompleteComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.0", type: FormAutocompleteComponent, selector: "quick-form-autocomplete", inputs: { options: "options", target: "target", control: "control" }, ngImport: i0, template: "<ul class=\"autocomplete-wrapper\" *ngIf=\"isVisible && options.length > 0\">\n    <li class=\"autocomplete-option\" *ngFor=\"let option of filteredOptions\" (click)=\"select(option)\">{{option}}</li>\n</ul>", styles: [":host{position:relative}ul{z-index:100;background-color:#fff;position:absolute;border:1px solid var(--gray-300);border-radius:.5rem;width:100%;overflow:hidden;overflow-y:scroll;max-height:200px}ul li{padding:.5rem;font-weight:500;color:var(--gray-700);cursor:pointer}ul li:hover{background-color:var(--primary-color);color:var(--text-primary-contrast)}\n"], dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.0", ngImport: i0, type: FormAutocompleteComponent, decorators: [{
            type: Component,
            args: [{ selector: 'quick-form-autocomplete', template: "<ul class=\"autocomplete-wrapper\" *ngIf=\"isVisible && options.length > 0\">\n    <li class=\"autocomplete-option\" *ngFor=\"let option of filteredOptions\" (click)=\"select(option)\">{{option}}</li>\n</ul>", styles: [":host{position:relative}ul{z-index:100;background-color:#fff;position:absolute;border:1px solid var(--gray-300);border-radius:.5rem;width:100%;overflow:hidden;overflow-y:scroll;max-height:200px}ul li{padding:.5rem;font-weight:500;color:var(--gray-700);cursor:pointer}ul li:hover{background-color:var(--primary-color);color:var(--text-primary-contrast)}\n"] }]
        }], propDecorators: { options: [{
                type: Input
            }], target: [{
                type: Input
            }], control: [{
                type: Input
            }] } });
